run_distillation.py

```
@@ -213,467 +232,603 @@ def parse_args_to_config() -> TrainingConfig:
+    def _configure_deterministic_mode(self) -> None:
+        # Optional deterministic mode
+        if getattr(self.config, "deterministic", False):
+            os.environ.setdefault("CUBLAS_WORKSPACE_CONFIG", ":4096:8")
+            torch.use_deterministic_algorithms(True)
+
+    def _configure_tmpdirs(self) -> None:
+        # Prefer node-local tmp to avoid NFS .nfs tombstones on cleanup
+        # If SLURM provides a local scratch (e.g., /dev/shm), use it; fall back to repo tmp
+        node_tmp = os.environ.get("TMPDIR")
+        if not node_tmp:
+            shm_candidate = f"/dev/shm/{os.environ.get('USER', 'user')}.sampledkd.{os.environ.get('SLURM_JOB_ID', 'local')}"
```

Ruff (0.14.1)

Probable insecure usage of temporary file or directory: "/dev/shm/"

(S108)